### PR TITLE
PIPE2D-810: Allow the user to specify a pfsConfig file to be used by mergeArms

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**
I decided to go with only adding the option to mergeArms

### CODE REVIEW

File: python/pfs/drp/stella/mergeArms.py [hunk 1]
```diff
@@ -39,6 +41,8 @@ class MergeArmsConfig(Config):
     doBarycentricCorr = Field(dtype=bool, default=True, doc="Do barycentric correction?")
     mask = ListField(dtype=str, default=["NO_DATA", "CR", "INTRP", "SAT"],
                      doc="Mask values to reject when combining")
+    pfsConfigFile = Field(dtype=str, default="", doc="""Full pathname of pfsCalib file to use.
+    If of the form "pfsConfig-0x%x-%d.fits", the pfsDesignId and visit0 will be deduced from the filename""")
```
Comment (hassanxp) [inline python/pfs/drp/stella/mergeArms.py:45]: If not of that form, the default pfsDesignId and visit0 values should be mentioned in the docString (I believe 666 and 0 respectively).

File: python/pfs/drp/stella/mergeArms.py [hunk 2]
```diff
@@ -151,7 +155,21 @@ def runDataRef(self, expSpecRefList):
...
+        if self.config.pfsConfigFile:
+            try:
+                pfsDesignId, visit0 = re.split(r"[-.]", os.path.split(self.config.pfsConfigFile)[1])[1:-1]
+
+                pfsDesignId = int(pfsDesignId, 16)
+                visit0 = int(visit0)
+            except ValueError:
+                pfsDesignId = 666
+                visit0 = 0
+
```
Comment (hassanxp) [inline python/pfs/drp/stella/mergeArms.py:167]: Is there a physical pfsDesign file available with that designId?

**Review by hassanxp — COMMENTED** [review]
Changes look OK, but need more understanding how the default values for ``designId`` and ``visit0`` are handled.

File: python/pfs/drp/stella/mergeArms.py (hunk 2 above)
Comment (RobertLuptonTheGood) [reply python/pfs/drp/stella/mergeArms.py:167]: In general not, but it doesn't actually matter.  The only reason for needing them is to make the pfsConfig I/O API happy

**Pull request merged**